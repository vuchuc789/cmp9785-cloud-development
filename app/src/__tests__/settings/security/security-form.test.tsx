import SecurityPage from '@/app/settings/security/page';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { act } from 'react';

jest.mock('next/navigation', () => ({
  useRouter: () => ({ replace: () => {} }),
}));

jest.mock('../../../contexts/auth', () => ({
  useAuth: jest.fn(() => ({ state: { accessToken: 'abc', isLoading: false } })),
}));

describe('SecurityPage', () => {
  it('renders profile page unchanged', async () => {
    let container;
    await act(async () => {
      const result = render(<SecurityPage />);

      container = result.container;
    });

    expect(container).toMatchSnapshot();
  });
});
