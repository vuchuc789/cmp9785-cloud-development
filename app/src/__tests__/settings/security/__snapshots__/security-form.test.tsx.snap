// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityPage renders profile page unchanged 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <div>
      <h3
        class="text-lg font-medium"
      >
        Security
      </h3>
      <p
        class="text-sm text-muted-foreground"
      >
        Update your password here.
      </p>
    </div>
    <div
      class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px"
      data-orientation="horizontal"
      data-slot="separator-root"
      role="none"
    />
    <form
      class="space-y-8"
    >
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="password"
        >
          Password
        </label>
        <input
          aria-describedby=":r0:-form-item-description"
          aria-invalid="false"
          autocomplete="new-password"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="password"
          name="password"
          type="password"
          value=""
        />
        <p
          class="text-muted-foreground text-sm"
          data-slot="form-description"
          id=":r0:-form-item-description"
        >
          Your new password.
        </p>
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="password_repeat"
        >
          Confirm password
        </label>
        <input
          aria-describedby=":r1:-form-item-description"
          aria-invalid="false"
          autocomplete="new-password"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="password_repeat"
          name="password_repeat"
          type="password"
          value=""
        />
        <p
          class="text-muted-foreground text-sm"
          data-slot="form-description"
          id=":r1:-form-item-description"
        >
          Confirm your new password.
        </p>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 cursor-pointer"
        data-slot="button"
        type="submit"
      >
        Update password
      </button>
    </form>
  </div>
</div>
`;
